From: Jiri Slaby <jslaby@suse.cz>
Date: Thu, 12 Feb 2026 07:25:14 +0100
Subject: Restore kABI for virtualbox drivers
References: bsc#1258115
Patch-mainline: Never, (re-)adding support for out-of-tree module

The symbols were unexported (effectively made private to KVM) in:
torvalds/linux@6276c67

virtualbox needs these symbols, otherwise the build fails:
  ERROR: modpost: module vboxdrv uses symbol cr4_read_shadow from namespace module:kvm,kvm-amd, but does not import it.
  ERROR: modpost: module vboxdrv uses symbol __flush_tlb_all from namespace module:kvm,kvm-amd, but does not import it.
  ERROR: modpost: module vboxdrv uses symbol cr4_update_irqsoff from namespace module:kvm,kvm-amd, but does not import it.

Export them, until vbox keeps up.

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 arch/x86/kernel/cpu/common.c | 4 ++--
 arch/x86/mm/tlb.c            | 2 +-
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/arch/x86/kernel/cpu/common.c b/arch/x86/kernel/cpu/common.c
index e7ab22fce3b5..539c19b26546 100644
--- a/arch/x86/kernel/cpu/common.c
+++ b/arch/x86/kernel/cpu/common.c
@@ -487,14 +487,14 @@ void cr4_update_irqsoff(unsigned long set, unsigned long clear)
 		__write_cr4(newval);
 	}
 }
-EXPORT_SYMBOL_FOR_KVM(cr4_update_irqsoff);
+EXPORT_SYMBOL_GPL(cr4_update_irqsoff);
 
 /* Read the CR4 shadow. */
 unsigned long cr4_read_shadow(void)
 {
 	return this_cpu_read(cpu_tlbstate.cr4);
 }
-EXPORT_SYMBOL_FOR_KVM(cr4_read_shadow);
+EXPORT_SYMBOL_GPL(cr4_read_shadow);
 
 void cr4_init(void)
 {
diff --git a/arch/x86/mm/tlb.c b/arch/x86/mm/tlb.c
index f5b93e01e347..f177df5c7046 100644
--- a/arch/x86/mm/tlb.c
+++ b/arch/x86/mm/tlb.c
@@ -1724,7 +1724,7 @@ void __flush_tlb_all(void)
 		flush_tlb_local();
 	}
 }
-EXPORT_SYMBOL_FOR_KVM(__flush_tlb_all);
+EXPORT_SYMBOL_GPL(__flush_tlb_all);
 
 void arch_tlbbatch_flush(struct arch_tlbflush_unmap_batch *batch)
 {
-- 
2.52.0

